### PR TITLE
firecracker: Network API update

### PR DESCRIPTION
**Pull request by mcastelino**
The Network API no longer supports the state field. Using the
state field causes the API call to fail.

Note: The firecracker binary needs to be updated to match the API
as the older versions will expect the state field.

Signed-off-by: Manohar Castelino <manohar.r.castelino@intel.com>

### CODE REVIEW

**Comment by caoruidong** [comment]
lgtm and /test

File: virtcontainers/fc.go [hunk 1]
```diff
@@ -525,7 +525,6 @@ func (fc *firecracker) fcAddNetDevice(endpoint Endpoint) error {
 		GuestMac:          endpoint.HardwareAddr(),
 		IfaceID:           &ifaceID,
 		HostDevName:       endpoint.NetworkPair().TapInterface.TAPIface.Name,
-		State:             "Attached",
```
Comment (bergwolf) [inline virtcontainers/fc.go]: I guess you need to update FC vendor for it as well?

Comment (mcastelino) [reply virtcontainers/fc.go]: @bergwolf Makes sense, even though that API change was already present before and I missed it as we were using an older firecracker binary. Revendored to match 0.12.0.

**Comment by bergwolf** [comment]
Thanks @mcastelino !
LGTM.

**Pull request merged**